### PR TITLE
[P3 tests] [Team 5] Add Task3: Compare Index with Compression

### PR DESCRIPTION
**Pull request by shaheming**
### What's your team number?
Team 5

### What is the functionality being tested?
Task3: Compare Index with Compression

### Describe your tests briefly:
- test1:  Test simple documents with same text, each key word show only one time each document mainly test inverted list since inverted list is long but positional list is short.
- test2:  Test docs with different text and each key word show multiple times in multiple document mainly test inverted list since inverted list is long but positional list is short.
- test3:  Test docs with different text and each key word show multiple times only in a document  mainly test positional  list since inverted  since  positional list is long.
- test4: Test add really long document, testing the positional list works or not mainly test positional  list since inverted  since  positional list is long.
### Does each test case have comments/documentation?
Yes

### CODE REVIEW

**Comment by Haixiang6123** [comment]
Test cases look good. And printing useful information makes it clear to understand your test cases. But it's always better to use other functions to encapsulate some codes.

**Comment by Haixiang6123** [comment]
Also, you can make a test case for testing empty text, empty documents.

**Comment by shaheming** [comment]
> Also, you can make a test case for testing empty text, empty documents.

```Java
    public Document(String text) {
        checkNotNull(text);
        this.text = text;
    }
```
This constructor will throw error, if we use the empty as imput.

File: src/test/java/edu/uci/ics/cs221/index/positional/Team5IndexCompressionTest.java [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+        InvertedIndexManager.createOrOpenPositional(path1, analyzer, naivecompressor);
+    positional_list_compressor =
+        InvertedIndexManager.createOrOpenPositional(path2, analyzer, compressor);
+  }
+
+  // test simple documents with same text, each key word show only one time each document
+  // mainly test inverted list since inverted list is long but positional list is short
+  @Test
+  public void Test1() {
+    Assert.assertEquals(0, PageFileChannel.readCounter);
```
Comment (NinCors) [inline src/test/java/edu/uci/ics/cs221/index/positional/Team5IndexCompressionTest.java]: Just one little advice, you can use the resetCounter function to reset the counter instead of throwing an error.

Comment (shaheming) [reply src/test/java/edu/uci/ics/cs221/index/positional/Team5IndexCompressionTest.java]: Screen Shot 2019-05-14 at 10 27 48 AM
Although we use assert there will throw exception, the @after Decorator will make sure that function will execute. In the after, we reset the counter. So, we do not need to worry about it.
